source/js/front/LikeModule.ts

```diff
@@ -0,0 +1,73 @@
+import { LikedPost } from "./likedPost";
+
+declare const wpApiSettings: any;
+
+class LikeModule {
+	constructor(
+        private postIds: LikedPost[],
+        private postTypesToShow: Array<string>,
+        private renderContainer: HTMLElement,
+        private noPostsNotice: HTMLElement,
+        private preLoaders: NodeListOf<HTMLElement>,
+
+    ) {
+            this.handleLikedPosts();
+    }
+
+	private handleLikedPosts(): void {
+        if (!wpApiSettings) {
+            console.log('wpApiSettings not found');
+            return;
+        }
+		const urlParams = new URLSearchParams(window.location.search);
+		const sharedPosts = urlParams.get('liked-posts');
+
+        this.fetchPosts();
+	}
+
+    private fetchPosts(): void {
```

Comment:
this.postIds and wpApiSettings would be nice to have as function parameters instead.